docs/operations/metrics.md

```diff
@@ -221,6 +221,41 @@ Note: If the JVM does not support CPU time measurement for the current thread, i
 |`worker/taskSlot/total/count`|Number of total task slots on the reporting worker per emission period. This metric is only available if the WorkerTaskCountStatsMonitor module is included.|category, version.|Varies.|
 |`worker/taskSlot/used/count`|Number of busy task slots on the reporting worker per emission period. This metric is only available if the WorkerTaskCountStatsMonitor module is included.|category, version.|Varies.|
 
+## Batch ingestion metrics (Native parallel task)
```

Comment:
Why did we decide to have a separate metric name for each type vs a single metric name with the type being a dimension?

`ingest/batch/count`, `ingest/batch/segments/count` and a dimension `type` (or even 2 dimensions for `isAppendToExisting` and `isDropExisting` - in this case, you don't need to deal with the error case described on line 243)

```
`true` | `true  ` | Invalid combination, exception thrown. |
```

Reply:
In a similar vein, won't these metrics be useful for streaming ingestion as well - Why not make the metric name `ingest/count` and then the `taskType` dimension tells you whether or not it is streaming or batch?

Also, reading these docs, it's not clear to me whether a compaction job will also be counted as a batch job

Reply:
The goal of this feature is to gather data to let us know the utilization of the various modes of batch ingest, in particular `REPLACE` thus the categorization.

Reply:
The invalid combination is enabled by the fact that the `IOConfig` part of the spec exposes both `dropExisting` and `appendToExisting` today. If we were to deprecate those and expose instead `batchProcessingMode` with its three possible values then this invalid state would no longer be possible. But I think that it is better to do this later if we agree it is a good idea.

Reply:
The `taskType` is already a dimension (i.e. for native batch types we have: `compact`, `index`, `index_parallel` and for streaming `index_kafka` , etc; for non native `index_hadoop`...) so yeah...maybe I should just drop the 'batch' from the metrics and just use the `indexType` dimension to filter...is this what you are thinking @suneet-s ?

Reply:
I agree with @suneet-s  suggestion to drop the batch and compact from metric name, as the task-type dimension included signifies this, and then the metric can be reused for other ingestion types, not just batch / compact.

Reply:
Yes, I agree too. Working on it now.

Reply:
> In a similar vein, won't these metrics be useful for streaming ingestion as well - Why not make the metric name ingest/count and then the taskType dimension tells you whether or not it is streaming or batch?

I had suggested a batch-specific name when discussing with Agustin since I think the semantics of the metric are batch specific, it's being emitted once per ingest spec posted, and I'm not sure what the equivalent for streaming would be. It's a metric indicating "number of times user did some action with a specific intent" vs. "detail of execution" (like a parallel subtask count for batch or number of streaming subtasks)

Reply:
I have removed the batch type from the metric and added `taskIngestionMode` as a dimension. As a result the new metrics are much clean and the code is cleaner as well. Also, now it was simple to add the metrics to streaming as well, take a look.